src/Logger/BadgeLogger.php

```diff
@@ -54,22 +55,25 @@ final class BadgeLogger implements MutationTestingResultsLogger
     private StrykerApiKeyResolver $strykerApiKeyResolver;
     private StrykerDashboardClient $strykerDashboardClient;
     private MetricsCalculator $metricsCalculator;
-    private string $branch;
+    private ?string $exactBranchMatch;
```

Comment:
IMO we could perhaps get rid of this long-term: for now, I kept it.

Internals could build a single regex when an `$exactBranchMatch` exists, but it's probably easier to proceed this way for now.

Reply:
Alternatively, `Branch` could provide a `->matches(string $branch): bool`, which would remove this state proliferation, and reduce test amount.

Reply:
In fact, I like `Badge#availableFor(string $branchName)`.

I'll probably refactor this, but **after** I have a first review from the maintainers :)